server/monitor-types/snmp.js

```diff
@@ -0,0 +1,107 @@
+                    if (error) {
+                        reject(error);
+                    } else {
+                        log.debug("monitor", `SNMP: Received varbinds: Type: ${getKey(snmp.ObjectType, varbinds[0].type)}, Value: ${varbinds[0].value}`); // Log the received varbinds for debugging
+                        resolve(varbinds);
+                    }
+                });
+            });
+
+            if (varbinds.length === 0 || getKey(snmp.ObjectType, varbinds[0].type) === 'NoSuchInstance') {
+                throw new Error(`No varbinds returned from SNMP session (OID: ${monitor.snmpOid})`);
+            } else {
+                const value = varbinds[0].value;
+                const numericValue = parseInt(value);
+                const stringValue = value.toString('ascii');
+
+                switch (monitor.snmpCondition) {
+                    case '>':
+                        heartbeat.status = numericValue > monitor.snmpControlValue ? UP : DOWN;
+                        break;
+                    case '>=':
+                        heartbeat.status = numericValue >= monitor.snmpControlValue ? UP : DOWN;
+                        break;
+                    case '<':
+                        heartbeat.status = numericValue < monitor.snmpControlValue ? UP : DOWN;
+                        break;
+                    case '<=':
+                        heartbeat.status = numericValue <= monitor.snmpControlValue ? UP : DOWN;
+                        break;
+                    case '==':
+                        if (!isNaN(value) && !isNaN(monitor.snmpControlValue)) {
+                            // Both values are numeric, parse them as numbers
+                            heartbeat.status = parseFloat(value) === parseFloat(monitor.snmpControlValue) ? UP : DOWN;
+                        } else {
+                            // At least one of the values is not numeric, compare them as strings
+                            heartbeat.status = value.toString() === monitor.snmpControlValue.toString() ? UP : DOWN;
+                        }
+                        break;
+                    case 'contains':
+                        heartbeat.status = stringValue.includes(monitor.snmpControlValue) ? UP : DOWN;
+                        break;
+                    default:
+                        heartbeat.status = DOWN;
+                        heartbeat.msg = `Invalid condition: ${monitor.snmpCondition}`;
+                        break;
+                }
+                heartbeat.msg = `SNMP value ` + (heartbeat.status ? `passes` : `does not pass`) + ` comparison: ${value.toString('ascii')} ${monitor.snmpCondition} ${monitor.snmpControlValue}`;
+
+            }
+            session.close(); // Close the session after use
+
+        } catch (err) {
+            heartbeat.status = DOWN;
+            heartbeat.msg = `SNMP Error: ${err.message}`;
+            log.debug("monitor", `SNMP: ${heartbeat.msg}`);
```

Comment:
no need to have `SNMP` twice.
Also unsure if this logging is needed/benefial. Are we not doing this in the general code? 

```suggestion
            log.debug("monitor", heartbeat.msg);
```

Reply:
I agree this log can come out. Resolved with 9ba0f68a866f4ba11602addbda362169af4b6907.